subprojects/docs/src/docs/userguide/migration/upgrading_version_5.adoc

```diff
@@ -42,8 +42,19 @@ Some plugins will break with this new version of Gradle, for example because the
 
 The usage of the `compile` and `runtime` configurations in the Java ecosystem plugins has been discouraged since https://docs.gradle.org/3.4/release-notes.html#the-java-library-plugin[Gradle 3.4].
 
-These configurations are used for compiling and running code from the `main` source set. Other sources sets create similar configurations (e.g. `testCompile` and `testRuntime` for the `test` source set), should not be used either.
-The `implementation`, `api`, `compileOnly` and `runtimeOnly` configurations should be used to declare dependencies and the `compileClasspath` and `runtimeClasspath` configurations to resolve dependencies.  See <<java_library_plugin.adoc#sec:java_library_configurations_graph, the relationship of these configurations>>.
+These configurations are used for compiling and running code from the `main` source set.
+Other sources sets create similar configurations (e.g. `testCompile` and `testRuntime` for the `test` source set), should not be used either.
+The `implementation`, `api`, `compileOnly` and `runtimeOnly` configurations should be used to declare dependencies and the `compileClasspath` and `runtimeClasspath` configurations to resolve dependencies.
+See <<java_library_plugin.adoc#sec:java_library_configurations_graph, the relationship of these configurations>>.
+
+==== Legacy publication system is deprecated and replaced with the `*-publish` plugins
+
+The `uploadArchives` task and the `maven` plugin are deprecated.
+
+Users should migrate to the <<publishing_setup.adoc#,publishing system>> of Gradle by using either the <<publishing_ivy.adoc#,`ivy-publish`>> or <<publishing_maven.adoc#,`maven-publish`>> plugins.
+These plugins have been stable since Gradle 4.8.
+
+The publishing system is also the only way to ensure the publication of https://blog.gradle.org/gradle-metadata-1.0[Gradle Module Metadata].
```

Comment:
I wonder if it is generally better to link to the docs on GMM - https://docs.gradle.org/current/userguide/publishing_gradle_module_metadata.html - here as it contains further links to more documentation (documentation was written after the blog post).